test/syscalls/call.js

```diff
@@ -0,0 +1,793 @@
+const debug = require('debug')
+const assert = require('assert')
+
+const Kernel = artifacts.require('./Kernel.sol')
+const abi = require('ethereumjs-abi')
+
+const beakerlib = require("../../beakerlib");
+
+// Valid Contracts
+const Valid = {
+    Adder: artifacts.require('test/valid/Adder.sol'),
+    Multiply: artifacts.require('test/valid/Multiply.sol'),
+    Divide: artifacts.require('test/valid/Divide.sol'),
+    SysCallTest: artifacts.require('test/valid/SysCallTest.sol'),
+    SysCallTestCall: artifacts.require('test/valid/SysCallTestCall.sol'),
+    FirstNestedCall: artifacts.require('test/valid/NestedCalls/FirstNestedCall.sol'),
+    SecondNestedCall: artifacts.require('test/valid/NestedCalls/SecondNestedCall.sol'),
+    ThirdNestedCall: artifacts.require('test/valid/NestedCalls/ThirdNestedCall.sol'),
+    FourthNestedCall: artifacts.require('test/valid/NestedCalls/FourthNestedCall.sol'),
+    FifthNestedCall: artifacts.require('test/valid/NestedCalls/FifthNestedCall.sol'),
+    SixthNestedCall: artifacts.require('test/valid/NestedCalls/SixthNestedCall.sol'),
+}
+
+const TestWrite = artifacts.require('test/TestWrite.sol');
+
+const Invalid = {
+    Simple: artifacts.require('test/invalid/Simple.sol')
+}
+
+contract('Kernel', function (accounts) {
+    describe('Call capability', function () {
+        const procName = "SysCallTestCall";
+        const bytecode = Valid.SysCallTestCall.bytecode;
+
+        describe('A()', function () {
+            const testProcName = "TestWrite";
+            const testBytecode = TestWrite.bytecode;
+            const functionSpec = "A()";
+            it('A() should succeed when given cap', async function () {
+                // This tests calls a test procedure which changes a storage
+                // value in the kernel from 3 to 356.
+                const kernel = await Kernel.new();
+
+                const cap1 = new beakerlib.WriteCap(0x8000,2);
```

Comment:
Test fails when `cap1` isn't passed to the `"SysCallTestCall"` procedure, but this shouldn't happen.

Reply:
It should absolutely fail when not providing cap1, because you are then trying to use the wrong capability. Why shouldn't it fail?

Reply:
I see, I would not then provide the same cap to both the caller and callee, since it's not clear from reading the test alone which cap is being used.